exoscale/api/compute.py

```diff
@@ -1313,62 +1406,97 @@ def update(
             None
         """
 
+        kwargs = {}
+
+        if name is not None:
+            kwargs["name"] = name
+
+        if description is not None:
+            kwargs["description"] = description
+
+        if instance_anti_affinity_groups is not None:
+            kwargs["anti-affinity-groups"] = [
+                {"id": i.id} for i in instance_anti_affinity_groups
+            ]
+
+        if instance_deploy_target is not None:
+            kwargs["deploy-target"] = {"id": instance_deploy_target.id}
+
+        if instance_elastic_ips is not None:
+            kwargs["elastic-ips"] = [{"id": i.id} for i in instance_elastic_ips]
+
+        if instance_enable_ipv6 is not None:
+            kwargs["ipv6-enabled"] = instance_enable_ipv6
+
+        if instance_prefix is not None:
+            kwargs["instance-prefix"] = instance_prefix
+
+        if instance_private_networks is not None:
+            kwargs["private-networks"] = [
+                {"id": i.id} for i in instance_private_networks
+            ]
+
+        if instance_security_groups is not None:
+            kwargs["security-groups"] = [{"id": i.id} for i in instance_security_groups]
+
+        if instance_ssh_key is not None:
+            kwargs["ssh-key"] = instance_ssh_key.name
+
+        if instance_template is not None:
+            kwargs["template"] = {"id": instance_template.id}
+
+        if instance_type is not None:
+            kwargs["instance-type"] = {"id": instance_type.id}
+
         instance_user_data_content = None
         if instance_user_data is not None:
             instance_user_data_content = b64encode(
                 bytes(instance_user_data, encoding="utf-8")
-            )
+            ).decode("ascii")
+            kwargs["user-data"] = instance_user_data_content
 
-        try:
-            self.compute.cs.updateInstancePool(
-                id=self.id,
-                zoneid=self.zone.id,
-                name=name,
-                description=description,
-                serviceofferingid=instance_type.id if instance_type else None,
-                templateid=instance_template.id if instance_template else None,
-                rootdisksize=instance_volume_size,
-                userdata=instance_user_data_content,
-            )
-        except CloudStackApiException as e:
-            raise APIException(e.error["errortext"], e.error)
+        if instance_volume_size is not None:
+            kwargs["disk-size"] = instance_volume_size
+
+        self.compute._v2_request_async(
+            "PUT",
+            "/instance-pool/" + self.id,
+            zone=self.zone.name,
+            json={**kwargs},
```

Comment:
This is equivalent to:

```suggestion
            json=kwargs,
```

Usually the name "kwargs" refers to arbitrary keyword arguments you'd pass or receive as part of a function call. "data" or "payload" may be less surprising to future readers.

Reply:
Thanks, fixed 👍